download_and_create_reference_datasets/v02/mito/utils.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import logging
+import json
+import tqdm
+import tempfile
+import os
+import zipfile
+import requests
+
+import hail as hl
+
+logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s', level='INFO')
+logger = logging.getLogger(__name__)
+
+contig_recoding={'1': 'chr1', '10': 'chr10', '11': 'chr11', '12': 'chr12', '13': 'chr13', '14': 'chr14', '15': 'chr15',
+ '16': 'chr16', '17': 'chr17', '18': 'chr18', '19': 'chr19', '2': 'chr2', '20': 'chr20', '21': 'chr21', '22': 'chr22',
+ '3': 'chr3', '4': 'chr4', '5': 'chr5', '6': 'chr6', '7': 'chr7', '8': 'chr8', '9': 'chr9', 'X': 'chrX', 'Y': 'chrY',
+ 'MT': 'chrM', 'NW_009646201.1': 'chr1'}
+
+
+def download_file(url, to_dir=tempfile.gettempdir(), verify=True):
+    if not (url and url.startswith(("http://", "https://"))):
+        raise ValueError("Invalid url: {}".format(url))
+
+    local_file_path = os.path.join(to_dir, os.path.basename(url.rstrip('/')))
+
+    if verify != False:
+        response = requests.head(url)
+        size = int(response.headers.get('Content-Length', '0'))
+        if os.path.isfile(local_file_path) and os.path.getsize(local_file_path) == size:
+            logger.info("Re-using {} previously downloaded from {}".format(local_file_path, url))
+            return local_file_path
+
+    is_gz = url.endswith(".gz") or url.endswith(".zip")
+    response = requests.get(url, stream=is_gz, verify=True if verify is None else verify)
+    input_iter = response if is_gz else response.iter_content()
+
+    logger.info("Downloading {} to {}".format(url, local_file_path))
+    input_iter = tqdm.tqdm(input_iter, unit=" data" if is_gz else " lines")
+
+    with open(local_file_path, 'wb') as f:
+        f.writelines(input_iter)
+
+    input_iter.close()
+
+    return local_file_path
+
+
+def convert_json2tsv(json_path):
```

Comment:
using the number 2 to mean to word "to" makes code a bit harder to read - just use `convert_json_to_tsv`, or even just `json_to_tsv`